manifests/service.pp

```diff
@@ -109,6 +109,14 @@
       content => template('logstash/logstash.yml.erb'),
     }
 
+    if defined(Class['java']) {
+      Class['java'] -> Exec['logstash-system-install']
```

Comment:
We shouldn't couple our module to an assumed `java` class. It could be anything. Maybe it's `puppetlabs/java` or maybe it's a site specific module that we don't know about.